### ISSUE TITLE
Userland tool to manage device on OpenWRT

### ISSUE THREAD

**Issue by bcdonadio** [sohaib17/mt7610u#1]
Hi!

I was able to build the module for OpenWRT 15.05 on the Raspberry Pi (bcm2708) with the SDK, and even did not get the RT2780STA.DAT loading error (I just copied the file to where it was asking it to be). However, iw can't work with the device.

My device is a TP-Link T2UH with the device id 148f:761a.

What userland tool did you use to interact (scan, connect, etc...) with the device?

**Comment by sohaib17** [comment]
@bcdonadio I too was able to load RT2780STA.DAT without any issues in Ubuntu 14.04 32-bit. I am guessing that OpenWRT 15.05 is also based on kernel version < 4.0. This issue appears with kernel version 4.0+.

I used OpenELEC connection manager to connect STA to an SSID. I had to enable WPA_SUPPLICANT for this though. After that I didn't get any chance to work on it again.

If I remember correctly, vendor documentation uses iwpriv to configure STA. I am not sure if OpenWRT has iwpriv since OpenELEC doesn't have iw tools by default.

**Comment by bcdonadio** [comment]
@sohaib17 Indeed, OpenWRT 15.05 uses Linux 3.18.

I will try to build the complete iw tools for this system and see what happens. Since ifconfig is able to bring up the ra0 interface, the only thing missing seems to be the command interface anyway.

As soon as I get any result, I report back.

Thanks!

**Comment by bcdonadio** [comment]
Soooo, got it to work! :D

I had to remove the autoload of the rt2800 and rt2x00 modules from OpenWRT by deleting their files on /etc/modules.d/, and then manually loading the mac80211 module before the mt7610u_sta.

After, I installed the wireless-tools package and connected with this command (open network, though, haven't tried WPA2 yet):

iwconfig ra0 mode Managed essid "My-ESSID" channel 1 ap AA:BB:CC:DD:EE:FF

I had to specify the AP mac address, otherwise it wouldn't connect.

Pretty happy, though. Thanks for the fixes!
